cmake/kokkos_tpls.cmake

```diff
@@ -81,6 +81,29 @@ kokkos_import_tpl(ROCTHRUST)
 if(Kokkos_ENABLE_DESUL_ATOMICS_EXTERNAL)
   find_package(desul REQUIRED COMPONENTS atomics)
   kokkos_export_cmake_tpl(desul REQUIRED COMPONENTS atomics)
+else()
+  set(test_16byte_atomic_code
+      "
+     #include<cstdint>
+     struct Dummy16Byte
+     {
+      int64_t a,b;
+     };
+     int main() {
+        Dummy16Byte val;
+        static_assert(__atomic_always_lock_free(sizeof(val),0),\"NO 128bit lock free atomics found\");
+       return 0;
+     }"
+  )
+
+  check_cxx_source_compiles("${test_16byte_atomic_code}" KOKKOS_HAVE_16BYTE_LOCK_FREE_ATOMICS_HOST)
```

Comment:
I think this will still need to get the host arch if it is set which it does not right now